spec/api/json_rpc_spec.rb

```diff
@@ -512,16 +613,115 @@ def wait_for_expect(retry_count = 20, sleep_duration = 0.5)
+                          "GroomDelta"
+                        ]
+                      },
+                      state: "none of the requested payloads match"
                     },
                     {
-                      mtype: 'exploit',
-                      mname: 'exploit/windows/smb/ms17_010_psexec',
+                      mname: "exploit/windows/smb/ms17_010_psexec",
+                      mtype: "exploit",
+                      options: {
+                        invalid: [],
+                        missing: [ "credential", "payload_match" ],
+                        required: [
+                          "RPORT",
+                          "SSLVersion",
+                          "ConnectTimeout",
+                          "DCERPC::max_frag_size",
+                          "DCERPC::ReadTimeout",
+                          "NTLM::UseNTLMv2",
+                          "NTLM::UseNTLM2_session",
+                          "NTLM::SendLM",
+                          "NTLM::UseLMKey",
+                          "NTLM::SendNTLM",
+                          "NTLM::SendSPN",
+                          "SMB::pipe_evasion",
+                          "SMB::pipe_write_min_size",
+                          "SMB::pipe_write_max_size",
+                          "SMB::pipe_read_min_size",
+                          "SMB::pipe_read_max_size",
+                          "SMB::pad_data_level",
+                          "SMB::pad_file_level",
+                          "SMB::obscure_trans_pipe_level",
+                          "SMBName", "SMB::VerifySignature",
+                          "SMB::ChunkSize", "SMB::Native_OS",
+                          "SMB::Native_LM",
+                          "SMB::AlwaysEncrypt",
+                          "SERVICE_PERSIST",
+                          "NAMED_PIPES",
+                          "LEAKATTEMPTS",
+                          "DBGTRACE",
+                          "Powershell::persist",
+                          "Powershell::prepend_protections_bypass",
+                          "Powershell::strip_comments",
+                          "Powershell::strip_whitespace",
+                          "Powershell::sub_vars",
+                          "Powershell::sub_funcs",
+                          "Powershell::exec_in_place",
+                          "Powershell::exec_rc4",
+                          "Powershell::remove_comspec",
+                          "Powershell::noninteractive",
+                          "Powershell::encode_final_payload",
+                          "Powershell::encode_inner_payload",
+                          "Powershell::wrap_double_quotes",
+                          "Powershell::no_equals",
+                          "Powershell::method",
+                          "SHARE",
+                          "ALLOW_GUEST"
+                        ]
+                      },
+                      state: "credentials are required, none of the requested payloads match"
```

Comment:
Shows that top level `state` would gate either on `REQUIRES_CRED` then expose `MISSING_PAYLOAD` or vice versa:
```suggestion
                      state: "REQUIRES_CRED",
                      detail: "credentials are required, none of the requested payloads match"
```